Lib/fontbakery/specifications/ufo_sources.py

```diff
@@ -0,0 +1,205 @@
+# ----------------------------------------------------------------------------
+
+
+@register_condition
+@condition
+def ufo_source(font):
+    import defcon
+    return defcon.Font(font)
+
+
+@register_check
+@check(id='com.daltonmaag/check/ufolint', priority=CRITICAL)
+def com_daltonmaag_check_ufolint(font):
+    """Run ufolint on UFO source directory."""
+    import subprocess
+    ufolint_cmd = ["ufolint", font]
+
+    try:
+        subprocess.check_output(ufolint_cmd, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        yield FAIL, ("ufolint failed the UFO source. Output follows :"
+                     "\n\n{}\n").format(e.output)
+    except OSError:
+        yield ERROR, "ufolint is not available!"
+
+    yield PASS, "ufolint passed the UFO source."
+
+
+@register_check
+@check(id='com.daltonmaag/check/required-fields')
+def com_daltonmaag_check_required_fields(font):
+    """Check that required fields are present in the UFO fontinfo.
+
+    ufo2ft requires these info fields to compile a font binary:
+    unitsPerEm, ascender, descender, xHeight, capHeight and familyName.
+    """
+    import defcon
+
+    recommended_fields = []
+    ufo = defcon.Font(font)
+
+    for field in [
+            "unitsPerEm", "ascender", "descender", "xHeight", "capHeight",
+            "familyName"
+    ]:
+        if ufo.info.__dict__.get("_" + field) is None:
+            recommended_fields.append(field)
+
+    if recommended_fields:
+        yield FAIL, "Required field(s) missing: {}".format(recommended_fields)
+    else:
+        yield PASS, "Required fields present."
+
+
+@register_check
+@check(id='com.daltonmaag/check/recommended-fields')
+def com_daltonmaag_check_recommended_fields(font):
+    """Check that recommended fields are present in the UFO fontinfo.
+
+    This includes fields that should be in any production font.
+    """
+    import defcon
+
+    recommended_fields = []
+    ufo = defcon.Font(font)
+
+    for field in [
+            "postscriptUnderlineThickness", "postscriptUnderlinePosition",
+            "versionMajor", "versionMinor", "styleName", "copyright",
+            "openTypeOS2Panose"
+    ]:
+        if ufo.info.__dict__.get("_" + field) is None:
+            recommended_fields.append(field)
+
+    if recommended_fields:
+        yield WARN, "Recommended field(s) missing: {}".format(
+            recommended_fields)
+    else:
+        yield PASS, "Recommended fields present."
+
+
+@register_check
+@check(id='com.daltonmaag/check/unnecessary-fields')
+def com_daltonmaag_check_unnecessary_fields(font):
+    """Check that no unnecessary fields are present in the UFO fontinfo.
+
+    ufo2ft will generate these.
+
+    openTypeOS2CodePageRanges is exempted because it is useful to toggle a
+    range when not _all_ the glyphs in that region are present.
+
+    year is deprecated since UFO v2.
+    """
+    import defcon
+
+    unnecessary_fields = []
+    ufo = defcon.Font(font)
```

Comment:
This is repeated in more than a single check. Make it a condition so that it can be reused (and probably save CPU time and memory)

You can look at specifications/googlefonts.py for examples such as this:
https://github.com/googlefonts/fontbakery/blob/fc9a4e68e5f002fe5df805b5fc6696b6bd1732cd/Lib/fontbakery/specifications/googlefonts.py#L78-L82

To use it, you simply pass the condition name as an argument to the check implementation.

Reply:
You don't need to use `conditions=['ufo']` in this case (just like in the ttFont, it is also a special case). But in general, for other conditions you may want to do this:

https://github.com/googlefonts/fontbakery/blob/fc9a4e68e5f002fe5df805b5fc6696b6bd1732cd/Lib/fontbakery/specifications/googlefonts.py#L591-L597

The result of doing so is that if registered_vendor_ids is not available ("if not True"), then the check will be automatically skipped and this will be logged in the fontbakery summary report.